org.eclipse.m2e.core/plugin.xml

```diff
@@ -22,6 +22,7 @@
    <extension-point id="conversionEnabler" name="%extension-point.conversionEnabler.name" schema="schema/conversionEnabler.exsd"/>
    <extension-point id="incrementalBuildFrameworks" name="%extension-point.incrementalBuildFrameworks.name" schema="schema/incrementalBuildFrameworks.exsd"/>
    <extension-point id="workspaceClassifierResolvers" name="%extension-point.workspaceClassifierResolvers.name" schema="schema/workspaceClassifierResolvers.exsd"/>
+   <extension-point id="m2eComponentSearch" name="m2e component search" schema="schema/m2eComponentSearch.exsd"/>
```

Comment:
What are the "core" cases that may make search useful here? I would expect the component search to be a pure-UI operation.

Reply:
> What are the "core" cases that may make search useful here? I would expect the component search to be a pure-UI operation.

At the moment there is also the [org.eclipse.m2e.sourcelookup](https://github.com/eclipse-m2e/m2e-core/tree/master/org.eclipse.m2e.sourcelookup) code where it may make sense to replace the existing code (which also talks to S.M.O). Once upon a time there was also the thought that some of the m2e bundles might be headless use cases.

Reply:
As mentioned in another comment, is the headless use-case a realistic one? It noticeably increases complexity and development/maintenance cost compared to make it fully about UI; so we shouldn't force such constraints unless we're sure of an ROI.

Reply:
Lemminx-maven would definitely benefit from this search. Ideally we'd make the lucene based index optional in m2e, as it brings too much pain ( too big to download, too many OOMEs, dead weight for jdt.ls)
So yeah headless based search is 100% needed

Reply:
Lemminx-Maven will never depend on m2e, nor use OSGi, nor the extension registry.
A search engine would be useful for Lemminx-Maven, but it would be misplaced in m2e. It would better be a separate plain Maven artifact, independent of m2e. So from both m2e and lemminx-maven perspective, having it headliess doesn't seem useful.